core/test/utils.hpp

```diff
@@ -90,23 +91,28 @@ using ComplexValueIndexTypes =
                      std::tuple<std::complex<double>, gko::int64>>;
 
 
-template <typename T>
+template <typename Precision, typename OutputType>
 struct reduction_factor {
-    static constexpr gko::remove_complex<T> value =
-        std::is_same<gko::remove_complex<T>, float>::value ? 1.0e-7 : 1.0e-14;
+    static constexpr remove_complex<OutputType> value =
+        std::is_same<gko::remove_complex<Precision>, gko::half>::value
+            ? 1.0e-2
+            : std::is_same<gko::remove_complex<Precision>, float>::value
+                  ? 1.0e-6
+                  : 1.0e-14;
```

Comment:
Can we get rid of these explicit numbers? specialization for `std::numeric_limits<gko::half>`?

Reply:
Sure, I can do that. It will likely be `10 * std::numeric_limits<T>` since we don't use the actual limits in our tests.